src/libraries/System.Runtime.Serialization.BinaryFormat/src/System/Runtime/Serialization/BinaryFormat/ClassRecord.cs

```diff
@@ -0,0 +1,182 @@
+    /// <exception cref="InvalidOperationException">Member of such name has value of a different type.</exception>
+    public ClassRecord? GetClassRecord(string memberName) => GetMember<ClassRecord>(memberName);
+
+    /// <inheritdoc cref="GetClassRecord(string)"/>
+    public string? GetString(string memberName) => GetMember<string>(memberName);
+    /// <inheritdoc cref="GetClassRecord(string)"/>
+    public bool GetBoolean(string memberName) => GetMember<bool>(memberName);
+    /// <inheritdoc cref="GetClassRecord(string)"/>
+    public byte GetByte(string memberName) => GetMember<byte>(memberName);
+    /// <inheritdoc cref="GetClassRecord(string)"/>
+    public sbyte GetSByte(string memberName) => GetMember<sbyte>(memberName);
+    /// <inheritdoc cref="GetClassRecord(string)"/>
+    public short GetInt16(string memberName) => GetMember<short>(memberName);
+    /// <inheritdoc cref="GetClassRecord(string)"/>
+    public ushort GetUInt16(string memberName) => GetMember<ushort>(memberName);
+    /// <inheritdoc cref="GetClassRecord(string)"/>
+    public char GetChar(string memberName) => GetMember<char>(memberName);
+    /// <inheritdoc cref="GetClassRecord(string)"/>
+    public int GetInt32(string memberName) => GetMember<int>(memberName);
+    /// <inheritdoc cref="GetClassRecord(string)"/>
+    public uint GetUInt32(string memberName) => GetMember<uint>(memberName);
+    /// <inheritdoc cref="GetClassRecord(string)"/>
+    public float GetSingle(string memberName) => GetMember<float>(memberName);
+    /// <inheritdoc cref="GetClassRecord(string)"/>
+    public long GetInt64(string memberName) => GetMember<long>(memberName);
+    /// <inheritdoc cref="GetClassRecord(string)"/>
+    public ulong GetUInt64(string memberName) => GetMember<ulong>(memberName);
+    /// <inheritdoc cref="GetClassRecord(string)"/>
+    public double GetDouble(string memberName) => GetMember<double>(memberName);
+    /// <inheritdoc cref="GetClassRecord(string)"/>
+    public decimal GetDecimal(string memberName) => GetMember<decimal>(memberName);
+    /// <inheritdoc cref="GetClassRecord(string)"/>
+    public TimeSpan GetTimeSpan(string memberName) => GetMember<TimeSpan>(memberName);
+    /// <inheritdoc cref="GetClassRecord(string)"/>
+    public DateTime GetDateTime(string memberName) => GetMember<DateTime>(memberName);
+
+    /// <returns>
+    /// <para>For primitive types like <seealso cref="int"/>, <seealso cref="string"/> or <seealso cref="DateTime"/> returns their value.</para>
+    /// <para>For nulls, returns a null.</para>
+    /// <para>For other types that are not arrays, returns an instance of <seealso cref="ClassRecord"/>.</para>
+    /// <para>For single-dimensional arrays returns <seealso cref="ArrayRecord{T}"/> where the generic type is the primitive type or <seealso cref="ClassRecord"/>.</para>
+    /// <para>For jagged and multi-dimensional arrays, returns an instance of <seealso cref="ArrayRecord"/>.</para>
+    /// </returns>
```

Comment:
TBH I would prefer to keep it, otherwise the returns block becomes very generic and not helpful (`GetRawValue` returns raw value, go to read the remarks to find out what it really means).

It does not look that bad:

![image](https://github.com/dotnet/runtime/assets/6011991/8a4739f7-7725-4b3c-b5d7-16dfd7367ae4)

Reply:
It looks like https://learn.microsoft.com/en-us/dotnet/api/system.type.equals?view=net-8.0#system-type-equals(system-object) used a bulleted list in their `<returns>`.  Not saying a bulleted list should be used here, but I guess there's precedent for a long <returns>.